### PR TITLE
Expose Connection Id in SignalR TS Client

### LINKED ISSUE #8681: Expose ConnectionId on SignalR Clients
We've had users ask about how to get a client's connectionId Our current guidance to users on how to get it is to create a Hub method to return it to them e.g.

```csharp
public string GetConnectionId() 
{
    return Context.ConnectionId
}

```
Which is less than ideal.

Another thing we planned on doing for the `onReconnected` events is making it have a single parameter which would be the client's new connectionId. 

- [x] .NET
- [x] Java
- [ ] JavaScript

### PR DESCRIPTION
**Pull request by mikaelm12**
Fixes #8681 
Last part of the exposing connection id issue.

### CODE REVIEW

File: src/SignalR/clients/ts/signalr/src/HubConnection.ts [hunk 1]
```diff
@@ -120,6 +120,17 @@ export class HubConnection {
         return this.connectionState;
     }
 
+    /** Represents the connection id of the {@link HubConnection} on the server. The connection id will be null when the connection is either
+     *  in the disconnected state or if the negotiation step was skipped.
+     */
+    get connectionId(): string | null {
+        if (this.connection.connectionId) {
```
Comment (davidfowl) [inline src/SignalR/clients/ts/signalr/src/HubConnection.ts]: Can connection be null?

Comment (JamesNK) [inline src/SignalR/clients/ts/signalr/src/HubConnection.ts]: Does `return this.connection.connectionId || null` work here? (I'm guessing the if is to go from string/undefined to string/null)

Comment (JamesNK) [reply src/SignalR/clients/ts/signalr/src/HubConnection.ts]: The property comment explains when connectionId could be null.

Comment (davidfowl) [reply src/SignalR/clients/ts/signalr/src/HubConnection.ts]: connection not connection property. If connection is null it'll null ref.

Comment (mikaelm12) [reply src/SignalR/clients/ts/signalr/src/HubConnection.ts]: The underlying connection should always be defined

Comment (mikaelm12) [reply src/SignalR/clients/ts/signalr/src/HubConnection.ts]: But we can add a check here if you want

Comment (davidfowl) [reply src/SignalR/clients/ts/signalr/src/HubConnection.ts]: We don't need it if it's never null

File: src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts [hunk 2]
```diff
@@ -838,6 +841,52 @@ describe("hubConnection", () => {
         }
     });
 
+    it("connection id matches server side connection id", async (done) => {
```
Comment (davidfowl) [inline src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts]: Do we have a test that verifies when it's null?

Comment (mikaelm12) [reply src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts]: Theres' an assert before start is called, there could be another for after stop. I'll add another test for when we skip negotiate and the id is always null.

**Review by davidfowl — APPROVED** [review]
